python/kiss_icp/config/parser.py

```diff
@@ -38,32 +39,23 @@ class KISSConfig(BaseSettings):
     data: DataConfig = DataConfig()
     mapping: MappingConfig = MappingConfig()
     adaptive_threshold: AdaptiveThresholdConfig = AdaptiveThresholdConfig()
-    _config_file: Optional[Path] = PrivateAttr()
-
-    def __init__(self, config_file: Optional[Path] = None, *args, **kwargs):
-        self._config_file = config_file
```

Comment:
Accessing an attribute before the parent's constructor was called is not possible with Pydantic V2. But even with the previous version, the pattern here is not considered very "pydantic" (see, e.g., [here](https://stackoverflow.com/a/65873630/250826)). Having the model class as plain as possible and filling it from the outside seems to be cleaner.

Reply:
Thanks for clarifying this!